### ISSUE TITLE
iOS production build/archive: Invalid bitcode signature issue

### ISSUE THREAD

**Issue by sunweiyang** [opentok/opentok-react-native#147]
In certain cases, iOS production build/archive can fail with the following error output:
```
Link /Users/sunw/Library/Developer/Xcode/DerivedData/Skale-fzshdwliplgfknhdofrjqmuxjbfz/Build/Intermediates.noindex/Skale.build/Release-iphoneos/Skale.build/Objects-normal/armv7/Skale normal armv7
error: Invalid bitcode signature
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

After a lot of digging, I've found that in OpenTokReactNative.xcodeproj's Build Settings, `Enable Bitcode` is set to `YES`. Changing this value to `NO` solved this issue for me.

<img width="1091" alt="screen shot 2018-08-06 at 2 38 41 pm" src="https://user-images.githubusercontent.com/12490625/43742637-84d89160-9987-11e8-9660-851b1accc1dc.png">

I'm happy to create a PR for this, but before I do that, @msach22 can you confirm that this wouldn't cause unintended downstream effects for other users? Thanks!

**Comment by msach22** [comment]
@sunweiyang Thanks for reporting this. Unfortunately, we should not limit the application to only work after when `enable bitcode` is set to `No` because that would not allow the Application to go through the App thinning process. 

I think I've discovered the fix for the library to work as expected:

Under `Build Settings` for the OpenTok React Native library target, go to `Strip Debug Symbols During Copy` and set Release to `Yes`. That should get the app to compile and archive as expected. You do not need to modify bit code settings for the app or the library.

Please let me know if you have any questions!

**Comment by HaroldoTeruya** [comment]
Thank you! I'll try this later and give a feedback.

**Comment by sunweiyang** [comment]
@msach22 thank you for the suggestion, but unfortunately setting `Strip Debug Symbols During Copy` as `Yes` did not fix the issue for me. So far, disabling bitcode is the only way that I can build my app for iOS.

**Comment by sunweiyang** [comment]
@msach22 Actually my codebase was accidentally using opentok-react-native@0.8.0 instead of 0.8.3 (yarn issue unrelated to this library), and once I re-installed our npm packages, this issue is mysteriously gone without any changes. Thank you for bearing with me, and apologies for the run-around.

@HaroldoTeruya, are you still experiencing an issue similar to this? If so, try making sure that you're on latest opentok-react-native, deleting your node_modules directory, and re-installing packages. I'll leave it up to either you or @msach22 to comment/close.

**Comment by sunweiyang** [comment]
By the way, I found that if I set OpenTokReactNative.xcodeproj's build target to `9.0` (instead of the current repo's value of `11.2`), I'm able to build 0.8.0 as well.

@msach22, perhaps it's a good idea for the repo to have `9.0` as the target to prevent future build issues?

**Comment by HaroldoTeruya** [comment]
Hi @sunweiyang and @msach22. It worked for me with the new version (0.8.3)!

Thank you @msach22 and @sunweiyang!

The following configurations is working for me:

I set the application target:
- `Enable Bitcode` to `Yes`;
- `Target Dependencies` the `OpenTokReactNative`;
- Added the `libOpenTokReactNative.a` in the `Link Binary With Libraries` ;

**Comment by msach22** [comment]
@HaroldoTeruya Glad to hear that it's working, which version of XCode are you using?

**Comment by HaroldoTeruya** [comment]
I am using the `Version 9.4.1`.

**Comment by msach22** [comment]
@HaroldoTeruya The empty swift file is not needed if you already have another swift file in the project. This is added so we can add `Swift` for the Swift flags to be enabled for the library.

**Comment by HaroldoTeruya** [comment]
@msach22 Understood!

After the last update, we can build in any device and emulator now.

But for archive I needed to:
My project -> Target -> Build Phases -> Target Dependencies -> (add the OpenTokReactNative).
Making this, the process to archive my entire project took more time than the normal. 

Is this an issue?

**Comment by msach22** [comment]
Hi @TeruyaHaroldo @sunweiyang Apologies for the delay, this was released with v0.9.0 last week.
